tunnel/intercept/iputils.go

```diff
@@ -66,10 +66,14 @@ func getInterceptIP(hostname string, resolver dns.Resolver) (net.IP, *net.IPNet,
 	if err == nil {
 		if len(addrs) > 0 {
 			ip := addrs[0].To4()
```

Comment:
I hit a case where two services where sharing an IP because I did the following

1. intercepted service A with hostname foo. It got address 100.64.0.3
2. Stopped intercepting service A
3. There may have been a tunneler restart in here, not sure
4. Intercepted service B with hostname bar. It got address 100.64.0.3
5. Intercepted service A with hostname foo. Foo still resolved to 100.64.0.3, so it also got the same IP address

This checks to make sure that a DNS resolve isn't coming out of our IP addr range, assuming that if it does we've got a state DNS entry. If that's detected we'll assign a new IP addr to avoid conflicts.